uniref-rest/src/main/java/org/uniprot/api/uniref/repository/store/UniRefEntryStoreRepository.java

```diff
@@ -163,17 +197,17 @@ private UniRefEntryLight getUniRefEntryLightFromStore(String clusterId) {
                 .orElseThrow(() -> new ResourceNotFoundException(CLUSTER_ID_NOT_FOUND + clusterId));
     }
 
-    private CursorPage getPage(UniRefIdRequest uniRefIdRequest, int memberCount) {
-        if (uniRefIdRequest.getSize() == null) { // set the default result size
-            uniRefIdRequest.setSize(defaultPageSize);
+    private CursorPage getPage(UniRefMemberRequest memberRequest, int memberCount) {
+        if (memberRequest.getSize() == null) { // set the default result size
+            memberRequest.setSize(defaultPageSize);
         }
-        return CursorPage.of(uniRefIdRequest.getCursor(), uniRefIdRequest.getSize(), memberCount);
+        return CursorPage.of(memberRequest.getCursor(), memberRequest.getSize(), memberCount);
     }
 
-    private List<String> getMembersPage(CursorPage page, List<String> members) {
+    private List<String> getMembersIdPage(CursorPage page, List<String> members) {
         int offset = page.getOffset().intValue();
         int nextOffset = CursorPage.getNextOffset(page);
-        return members.subList(offset, nextOffset);
+        return getMemberIds(members.subList(offset, nextOffset));
```

Comment:
Thank you, filter tests are in UniRefEntryFacetConfigTest and also there are Integration Tests in UniRefMembersControllerIT. Would you mind have a look and see if we are missing any use case, because this is a very important logic in this use case.

Reply:
yep i'll take a look